api/sirsidynix_authentication_provider.py

```diff
@@ -179,32 +162,24 @@ def remote_patron_lookup(
         if not data or "fields" not in data:
             return None
 
+        patrondata.complete = True
         fields: dict = data["fields"]
         patrondata.personal_name = fields.get("displayName")
         patron_type: str = fields["patronType"].get("key", "")
-
-        patrondata.external_type = patron_type
+        patrondata.library_identifier = patron_type
 
         # Basic block reasons
 
-        # Does the patron type start with the library prefix
-        # This ensures the patron is a member of the library
-        # they are interacting with
-        if not patron_type.startswith(self.sirsi_library_prefix):
-            # We respond with a problem detail, stopping the upstream authentication immediately
-            return PATRON_OF_ANOTHER_LIBRARY
-
         if not fields.get("approved", False):
             patrondata.block_reason = SirsiBlockReasons.NOT_APPROVED
             return patrondata
 
-        # Remove the library prefix from the patron type
-        # we are left with the patron "suffix"
-        # This suffix can be part of the blocked patron types list
-        patron_suffix = patron_type[len(self.sirsi_library_prefix) :]
-        if patron_suffix in self.sirsi_disallowed_suffixes:
-            patrondata.block_reason = SirsiBlockReasons.PATRON_BLOCKED
-            return patrondata
+        # If the patron type ends with a disallowed suffix the
+        # patron will be authenticated but marked as blocked.
+        for suffix in self.sirsi_disallowed_suffixes:
+            if patron_type.endswith(suffix):
+                patrondata.block_reason = SirsiBlockReasons.PATRON_BLOCKED
+                return patrondata
```

Comment:
An issue with not using the "prefix" is that we may have a prefix of "ca" and a disallowed suffix of "ad" which means "caad" should be disallowed.
Theoretically we may encounter a patron type of "cad" (suffix "d") which should be allowed, but will be blocked through this logic.
If this system is getting confusing we can just change this to DISALLOWED PATRON TYPES, wherein the entire type is explicitly disallowed eg. the i-team would input the entire "caad" as a disallowed type avoiding any confusion.

Reply:
That is a good point @RishiDiwanTT. I don't think we have any cases like this currently, but I'll have to go back over the spreadsheet of types to check and be sure.